src/server/scripts/Outland/BlackTemple/black_temple.cpp

```diff
@@ -178,7 +178,22 @@ class npc_wrathbone_flayer : public CreatureScript
     }
 };
 
+class go_den_of_mortal_door : public GameObjectScript
+{
+public:
+    go_den_of_mortal_door() : GameObjectScript("go_den_of_mortal_door") { }
+
+    void OnGameObjectStateChanged(GameObject* go, uint32 state) override
+    {
+        if (state == GO_STATE_ACTIVE)
+            if(InstanceScript* instance = go->GetInstanceScript())
+                if (Creature* trigger = instance->GetCreature(DATA_BLACK_TEMPLE_TRIGGER))
+                    trigger->AI()->Talk(EMOTE_DEN_OF_MORTAL_DOOR_OPEN);
```

Comment:
Trigger this from the instance script where the gameobject is activated instead, 1 less gameobject script.